src/connectors/configure-related-items/connectConfigureRelatedItems.ts

```diff
@@ -0,0 +1,157 @@
+  /**
+   * Function called to transform the search parameters generated.
+   */
+  transformSearchParameters?(
+    searchParameters: SearchParameters
+  ): PlainSearchParameters;
+}
+
+export type ConfigureRelatedItemsWidgetFactory<
+  TConfigureRelatedItemsWidgetParams
+> = WidgetFactory<
+  ConfigureRelatedItemsConnectorParams & TConfigureRelatedItemsWidgetParams
+>;
+
+type ConfigureRelatedItemsConnector = <TConfigureRelatedItemsWidgetParams>(
+  render?: ConfigureRenderer<ConfigureConnectorParams>,
+  unmount?: Unmounter
+) => ConfigureRelatedItemsWidgetFactory<TConfigureRelatedItemsWidgetParams>;
+
+const withUsage = createDocumentationMessageGenerator({
+  name: 'configure-related-items',
+  connector: true,
+});
+
+function createOptionalFilter({
+  attributeName,
+  attributeValue,
+  attributeScore,
+}) {
+  return `${attributeName}:${attributeValue}<score=${attributeScore || 1}>`;
+}
+
+const connectConfigureRelatedItems: ConfigureRelatedItemsConnector = (
+  renderFn,
+  unmountFn
+) => {
+  warning(
+    false,
+    'ConfigureRelatedItems is an experimental widget that is subject to change in next minor versions.'
+  );
+
+  return widgetParams => {
+    const { hit, matchingPatterns, transformSearchParameters = x => x } =
+      widgetParams || ({} as typeof widgetParams);
+
+    if (!hit) {
+      throw new Error(withUsage('The `hit` option is required.'));
+    }
+
+    if (!matchingPatterns) {
+      throw new Error(withUsage('The `matchingPatterns` option is required.'));
+    }
+
+    const optionalFilters = Object.keys(matchingPatterns).reduce<
+      Array<string | string[]>
+    >((acc, attributeName) => {
+      const attribute = matchingPatterns[attributeName];
+      const attributeValue = hit[attributeName];
+      const attributeScore = attribute.score;
+
+      if (Array.isArray(attributeValue)) {
+        return [
+          ...acc,
+          attributeValue.map(attributeSubValue => {
+            return createOptionalFilter({
+              attributeName,
+              attributeValue: attributeSubValue,
+              attributeScore,
+            });
+          }),
+        ];
+      }
+
+      if (typeof attributeValue === 'string') {
+        return [
+          ...acc,
+          createOptionalFilter({
+            attributeName,
+            attributeValue,
+            attributeScore,
+          }),
+        ];
+      }
+
+      warning(
+        false,
+        `
+The \`matchingPatterns\` option returned a value of type \`${typeof attributeValue}\` for the "${attributeName}" key. This value was not sent to Algolia because it's not supported by \`optionalFilters\`.
+
+You can remove the "${attributeName}" key from the \`matchingPatterns\` option.
+
+See https://www.algolia.com/doc/api-reference/api-parameters/optionalFilters/
+            `
+      );
+
+      return acc;
+    }, []);
+
+    const searchParameters: PlainSearchParameters = {
+      ...transformSearchParameters(
```

Comment:
Since we have no API for modifying props, we can indeed assume this all is static, but that will need to be rethought once we have an update api

Reply:
So like most of the connectors.